src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMDecoy.h

```diff
@@ -132,44 +138,70 @@ namespace OpenMS
       identity_threshold.
     */
     OpenMS::TargetedExperiment::Peptide shufflePeptide(
-      OpenMS::TargetedExperiment::Peptide peptide, double identity_threshold, int seed = -1,
-      int max_attempts = 100);
+      OpenMS::TargetedExperiment::Peptide peptide, const double identity_threshold, int seed = -1,
+      const int max_attempts = 100);
 
     /**
       @brief Pseudo-reverse a peptide sequence (with its modifications)
 
-      Pseudo reverses a peptide sequence, leaving the last AA constant
+      @note Pseudo reverses a peptide sequence, leaving the C terminus (the
+            last AA constant)
     */
     OpenMS::TargetedExperiment::Peptide pseudoreversePeptide(
-      OpenMS::TargetedExperiment::Peptide peptide);
+      const OpenMS::TargetedExperiment::Peptide& peptide) const;
 
     /**
       @brief Reverse a peptide sequence (with its modifications)
+
+      @note Does not keep N / C terminus in place.
     */
     OpenMS::TargetedExperiment::Peptide reversePeptide(
-      OpenMS::TargetedExperiment::Peptide peptide);
+      const OpenMS::TargetedExperiment::Peptide& peptide) const;
+
+    /**
+      @brief Reverse a peptide sequence (with its modifications)
+
+      @param peptide The peptide sequence and modifications
+      @param keepN Whether to keep N terminus in place
+      @param keepC Whether to keep C terminus in place
+      @param const_pattern A set of AA to leave in place
+    */
+    static OpenMS::TargetedExperiment::Peptide reversePeptide(
+                const OpenMS::TargetedExperiment::Peptide& peptide,
+                const bool keepN,
+                const bool keepC, 
+                const std::vector<String> & const_pattern = std::vector<String>());
 
     /**
       @brief Find all K, R, P sites in a sequence to be set as fixed
 
       This method was adapted from the SpectraST decoy generator
     */
-    std::vector<std::pair<std::string::size_type, std::string> > findFixedResidues(
-      std::string sequence);
+    IndexType findFixedResidues(std::string sequence);
```

Comment:
could this be
`IndexType findFixedResidues(const std::string& sequence) const;`?